### PR TITLE
Add endpoint for getting worker container logs

### PR DESCRIPTION
**Pull request by kiendang**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/worker/worker_pool_service.py [hunk 1]
```diff
@@ -179,3 +177,96 @@ def delete_worker(
...
+            return worker_pool_worker
+
+        _, worker = worker_pool_worker
+
+        docker_container = _get_worker_container(worker)
+        if isinstance(docker_container, SyftError):
+            return docker_container
+
+        try:
+            return docker_container.logs()
```
Comment (kiendang) [inline packages/syft/src/syft/service/worker/worker_pool_service.py]: This returns `bytes`, not `str`. I opted to just return it as is here. Users can `decode()` to string if needed.

File: packages/syft/src/syft/service/worker/worker_pool_service.py [hunk 2]
```diff
@@ -179,3 +177,99 @@ def delete_worker(
...
+        path="worker_pool.worker_logs",
+        name="worker_logs",
+        roles=DATA_OWNER_ROLE_LEVEL,
+    )
+    def worker_logs(
+        self,
+        context: AuthedServiceContext,
+        worker_pool_id: UID,
+        worker_id: UID,
+        raw: bool = False,
```
Comment (kiendang) [inline packages/syft/src/syft/service/worker/worker_pool_service.py:191]: @rasswanth-s added this argument. By default the endpoint returns `str` logs (`.decode()`). Users can opt to get the raw `bytes` logs if they want.

File: packages/syft/src/syft/service/worker/worker_pool_service.py [hunk 3]
```diff
@@ -131,26 +136,19 @@ def delete_worker(
         worker_id: UID,
         force: bool = False,
     ) -> Union[SyftSuccess, SyftError]:
-        worker_pool = self.stash.get_by_uid(
-            credentials=context.credentials, uid=worker_pool_id
+        worker_pool_worker = self._get_worker_pool_and_worker(
```
Comment (rasswanth-s) [inline packages/syft/src/syft/service/worker/worker_pool_service.py:139]: 👌

File: packages/syft/src/syft/service/worker/worker_pool_service.py [hunk 4]
```diff
@@ -179,3 +177,99 @@ def delete_worker(
...
+    return (
+        worker
+        if worker is not None
+        else SyftError(
+            message=f"Worker with id: {worker_id} not found in pool: {worker_pool.name}"
+        )
+    )
+
+
+def _get_worker_container(
```
Comment (rasswanth-s) [inline packages/syft/src/syft/service/worker/worker_pool_service.py:263]: 🙌

**Review by rasswanth-s — APPROVED** [review]
Well done @kiendang

**Comment by rasswanth-s** [comment]
Merging the PR ,as it failing due to the flaky mongoDB tests.

**Pull request merged**